packages/omgx/contracts/deploy/002-ERC20.deploy.ts

```diff
@@ -31,44 +30,49 @@ const deployFn: DeployFunction = async (hre) => {
     true,
   )
 
-  //Mint a new token on L1 and set up the L1 and L2 infrastructure
-  // [initialSupply, name, symbol]
-  // this is owned by bobl1Wallet
-  L1ERC20 = await Factory__L1ERC20.deploy(
-    initialSupply,
-    tokenName,
-    tokenSymbol
-  )
-  await L1ERC20.deployTransaction.wait()
-  const L1ERC20DeploymentSubmission: DeploymentSubmission = {
-    ...L1ERC20,
-    receipt: L1ERC20.receipt,
-    address: L1ERC20.address,
-    abi: L1ERC20Json.abi,
-  };
-  await hre.deployments.save('L1ERC20', L1ERC20DeploymentSubmission)
-  console.log(`🌕 ${chalk.red('L1ERC20 deployed to:')} ${chalk.green(L1ERC20.address)}`)
+  for (let token of preSupportedTokens.supportedTokens) {
+    //Mint a new token on L1 and set up the L1 and L2 infrastructure
+    // [initialSupply, name, symbol]
+    // this is owned by bobl1Wallet
+    L1ERC20 = await Factory__L1ERC20.deploy(
```

Comment:
for deployments to rinkeby, the other tokens except JLKN from `preSupportedTokens` already exists and we have to connect with these. same for mainnet, so let's not redeploy these l1 tokens in scripts, such that we could use these scripts for rinkeby as well

I see for local depl though, we would need these dummy tokens, but does presupport make sense in local( since all these tokens don't exist locally)? If we need, lets make the deployment conditional maybe?

Reply:
Added a new NETWORK flag.

Reply:
Tested, looks good - great point @souradeep-das - good fix @cby3149.